### DISCUSSION TITLE
Is there a way to add more tools available in proto into the `toolchain.yaml` config?

### DISCUSSION THREAD

**Discussion by rhuanbarreto** [moonrepo/moon#1769] [Ideas]
One of the great things about moon is that all the workspace setup with regards to the toolchain, like runtime installation (bun/node) and package installation is taken care of. This speeds up a lot a developer once they don't need to install things manually and it's "platform agnostic" (Mac/Linux).

What I'm missing is the ability to auto install underlying tools that you can install using proto, like the github cli.

Is there a way to describe those tools in toolchain.yaml and have them installed if required by tasks? I have some wrapper cli packages that uses those cli's underlying, like the Github and the Azure one.

Any ideas would be greatly appreaciated.

**Comment by milesj** [comment]
There's nothing built-in at the moment, but the migration to plugins would easily allow this.

**Comment by harlequin** [comment]
@rhuanbarreto we are facing the same issues and questions. 
Can you have a look into #1793 if this is what you are also have in mind?
